scripts/generate/public_source_gen.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Copyright (c) F5, Inc.
+# This source code is licensed under the Apache License, Version 2.0 license found in the
+# LICENSE file in the root directory of this source tree.
+
+# shellcheck shell=bash
+# Generate support file from a public git repo
+# Only call it through //go:generate comments in project root to update .gen.go files
+
+set -e
+
+branch="master"
+url=""
+output_path=""
+sub_path=""
+config_path=""
+
+help() {
+    cat << EOF
+Generate support file from a public repo(currently only git).
```

Comment:
What about an existing directory on disk?

Reply:
I thought user should run `./cmd/generate` directly in this situation?

Reply:
That could be fine